### PR TITLE
Task.yield() example

### PR DESCRIPTION
**Pull request by adriansergheev**
It seems to be a good place to have an example for `Task.yield()` but if I have misunderstood it's purpose or it should be moved into a fully separate sample code, please let me know!

[source](https://developer.apple.com/documentation/swift/task/yield())

### CODE REVIEW

**Comment by FranzBusch** [comment]
I think `Task.yield` is interesting and has very rare use-cases. It's probably good if we define those rare use-cases specifically.

File: Guide.docc/IncrementalAdoption.md [hunk 1]
```diff
@@ -508,3 +508,21 @@ await withTaskGroup(of: Something.self) { group in
...
+    await dependency.fetch()
+    // execute part of long running synchronous code
+    await Task.yield()  // explicitly insert a suspension point
+    // continue long running synchronous execution
+  }
+}
+```
+
+Introducing an explicit suspension point helps Swift balance between making progress on this task, and letting other tasks in your program make progress on their work.
+However, if this task has the highest-priority in the system, the executor immediately resumes execution of the same task.
```
Comment (adriansergheev) [inline Guide.docc/IncrementalAdoption.md:527]: is it enough to say `the system` here, or is it an execution context more appropriate?

**Comment by adriansergheev** [comment]
> I think `Task.yield` is interesting and has very rare use-cases. It's probably good if we define those rare use-cases specifically.

i agree, the previous example did not unpack enough about Task.yield() to warrant having it here (maybe this one doesn't either!)
i do have a hunch that it might be a good reference for people, even though "long running synchronous work" is not that common
i am also not sure how to fit another example from scratch in the narrative, so i tried rewriting the previous 🙂
otherwise [this](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/concurrency/) and [this](https://developer.apple.com/documentation/swift/task/yield()) explain yielding just fine!

**Comment by FranzBusch** [comment]
> > I think `Task.yield` is interesting and has very rare use-cases. It's probably good if we define those rare use-cases specifically.
> 
> i agree, the previous example did not unpack enough about Task.yield() to warrant having it here (maybe this one doesn't either!) i do have a hunch that it might be a good reference for people, even though "long running synchronous work" is not that common i am also not sure how to fit another example from scratch in the narrative, so i tried rewriting the previous 🙂 otherwise [this](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/concurrency/) and [this](https://developer.apple.com/documentation/swift/task/yield()) explain yielding just fine!

I do agree that the API level docs here are good. In my mind there is very little usage for `Task.yield` nowadays except the potential case of long-running synchronous work. However, even in this case using task executor preference is probably preferable since it moves the long running work into an isolated queue/thread.

**Review by mattmassicotte — APPROVED** [review]
Thank you so much for this!

File: Guide.docc/IncrementalAdoption.md (hunk 1 above)
Comment (mattmassicotte) [reply Guide.docc/IncrementalAdoption.md:527]: System could be interpreted to mean "whole process" I guess. But I think the overall idea is clear enough for now.

**Pull request merged**